API/Services/TachiyomiService.cs

```diff
@@ -0,0 +1,134 @@
+                    //Number = $"{int.Parse(volumeChapter.Number) / 1000f}"
+                };
+            }
+
+            var lastChapter = looseLeafChapterVolume.Chapters.OrderBy(c => float.Parse(c.Number), ChapterSortComparer.Default).Last();
+            return _mapper.Map<ChapterDto>(lastChapter);
+        }
+
+        // There is progress, we now need to figure out the highest volume or chapter and return that.
+        var prevChapter = await _unitOfWork.ChapterRepository.GetChapterDtoAsync(prevChapterId);
+        var volumeWithProgress = await _unitOfWork.VolumeRepository.GetVolumeDtoAsync(prevChapter.VolumeId, userId);
+        // We only encode for single-file volumes
+        if (volumeWithProgress.Number != 0 && volumeWithProgress.Chapters.Count == 1)
+        {
+            // The progress is on a volume, encode it as a fake chapterDTO
+            return new ChapterDto()
+            {
+                // Use R to ensure that localization of underlying system doesn't affect the stringification
+                // https://docs.microsoft.com/en-us/globalization/locale/number-formatting-in-dotnet-framework
+                Number = (volumeWithProgress.Number / 10000f).ToString("R", _englishCulture)
+
+            };
+        }
+
+        // Progress is just on a chapter, return as is
+        return prevChapter;
+    }
+
+    public async Task<bool> MarkChaptersUntilAsRead(AppUser userWithProgress, int seriesId, float chapterNumber)
+    {
+        userWithProgress.Progresses ??= new List<AppUserProgress>();
+
+        switch (chapterNumber)
+        {
+            // When Tachiyomi sync's progress, if there is no current progress in Tachiyomi, 0.0f is sent.
+            // Due to the encoding for volumes, this marks all chapters in volume 0 (loose chapters) as read.
+            // Hence we catch and return early, so we ignore the request.
+            case 0.0f:
+                return true;
+            case < 1.0f:
+            {
+                // This is a hack to track volume number. We need to map it back by x100
+                var chapterString = $"{chapterNumber}";
+                var volumeNumber = int.Parse($"{(int)(chapterNumber * 10000)}", _englishCulture);
+                //var volumeNumber = int.Parse($"{float.Parse(chapterString.Substring(0, Math.Min(7, chapterString.Length))) * 100f}");
+                await _readerService.MarkVolumesUntilAsRead(userWithProgress, seriesId, volumeNumber);
+                break;
+            }
+            default:
+                await _readerService.MarkChaptersUntilAsRead(userWithProgress, seriesId, chapterNumber);
+                break;
+        }
+
+
+        _unitOfWork.UserRepository.Update(userWithProgress);
+
+        if (!_unitOfWork.HasChanges()) return true;
+        if (await _unitOfWork.CommitAsync()) return true;
+
+        await _unitOfWork.RollbackAsync();
```

Comment:
We need a try catch so you can catch the exception and log it out on Error(ex, "There was an issue updating...")